src/app/bungie-api/bungie-service-helper.ts

```diff
@@ -36,226 +41,181 @@ const notifyTimeout = _.throttle(
+        TIMEOUT,
+        notifyTimeout
+      ),
+      API_KEY,
+      true
+    )
+  )
+);
+/** used to get manifest and global alerts*/
+export const unauthenticatedHttpClient = dimErrorHandledHttpClient(
+  responsivelyThrottleHttpClient(
+    createHttpClient(
+      createFetchWithNonStoppingTimeout(fetch, TIMEOUT, notifyTimeout),
+      API_KEY,
+      false
+    )
+  )
+);
 
 /** Generate an error with a bit more info */
-export function error(message: string, errorCode: PlatformErrorCodes): DimError {
+export function dimError(message: string, errorCode: PlatformErrorCodes): DimError {
   const error: DimError = new Error(message);
   error.code = errorCode;
   return error;
 }
 
-export async function handleErrors<T>(response: Response): Promise<ServerResponse<T>> {
-  if (response instanceof DOMException && response.name === 'AbortError') {
-    throw new Error(
-      navigator.onLine ? t('BungieService.SlowResponse') : t('BungieService.NotConnected')
-    );
-  }
+/**
+ * wrap HttpClient in handling specific to DIM, using i18n strings, bounce to login, etc
+ */
+export function dimErrorHandledHttpClient(httpClient: HttpClient): HttpClient {
+  return async (config: HttpClientConfig) => {
+    try {
+      return await httpClient(config);
+    } catch (e) {
+      handleErrors(e);
+    }
+  };
+}
 
-  if (response instanceof TypeError) {
+/**
+ * if HttpClient throws an error (js, Bungie, http) this enriches it with DIM concepts and then re-throws it
+ */
+export function handleErrors(error: Error) {
+  if (error instanceof DOMException && error.name === 'AbortError') {
     throw new Error(
-      navigator.onLine ? t('BungieService.NotConnectedOrBlocked') : t('BungieService.NotConnected')
+      navigator.onLine ? t('BungieService.SlowResponse') : t('BungieService.NotConnected')
     );
   }
 
-  if (response instanceof Error) {
-    throw response;
+  if (error instanceof SyntaxError) {
+    console.error('Error parsing Bungie.net response', error);
+    throw new Error(t('BungieService.Difficulties'));
   }
 
-  if (response.status === -1) {
+  if (error instanceof TypeError) {
     throw new Error(
       navigator.onLine ? t('BungieService.NotConnectedOrBlocked') : t('BungieService.NotConnected')
     );
   }
 
-  let data: ServerResponse<any> | undefined;
-  try {
-    data = await response.json();
-  } catch (e) {
-    if (e instanceof SyntaxError) {
-      console.error('Error parsing Bungie.net response', e);
-      throw new Error(t('BungieService.Difficulties'));
+  if (error instanceof HttpStatusError) {
+    // "I don't think they exist" --Westley, The Princess Bride (1987)
+    if (error.status === -1) {
+      throw new Error(
+        navigator.onLine
+          ? t('BungieService.NotConnectedOrBlocked')
+          : t('BungieService.NotConnected')
+      );
+    }
+
+    // Token expired and other auth maladies
+    if (error.status === 401 || error.status === 403) {
+      goToLoginPage();
```

Comment:
Note for the future, I'd love to extract this somehow, but I haven't quite figured out what the right pattern would be.